### PR TITLE
[DEV][BACKEND] Issue 55:  Sending Invitation

### PR DESCRIPTION
**Pull request by ynsmrkrkmz**
invitation controller, service and repository added
invitation model added
Invitationdto and InvitationCreateRequestDto added
Local profile properties added
data.sql queries modified for existing data

### CODE REVIEW

**Review by Oguzoz1 — APPROVED** [review]

File: Communitter/src/main/resources/application-local.properties
Comment (niloaydin) [inline Communitter/src/main/resources/application-local.properties:1]: can we please not push the changes for properties file

**Review by niloaydin — CHANGES_REQUESTED** [review]

File: Communitter/src/main/java/com/communitter/api/service/InvitationService.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    public InvitationDto inviteUser(User authUser, InvitationCreateRequestDto request) {
+
+        Community community = communityRepository.getReferenceById(request.getCommunityId());
+
+        if (community.isPublic()) {
+            throw new RuntimeException("Cannot invite a user for a public community");
+        }
+
+        User invitedUser = userRepository.findByUsername(request.getUsername()).orElseThrow();
```
Comment (niloaydin) [inline Communitter/src/main/java/com/communitter/api/service/InvitationService.java:39]: Can we throw specific messages. Also we can add try catch block to service and handle it in the controller

File: Communitter/src/main/resources/application-local.properties
Comment (ynsmrkrkmz) [reply Communitter/src/main/resources/application-local.properties:1]: the file is actually intentionally pushed for local development settings. you can use this profile without changing production one.

File: Communitter/src/main/java/com/communitter/api/service/InvitationService.java (hunk 1 above)
Comment (ynsmrkrkmz) [reply Communitter/src/main/java/com/communitter/api/service/InvitationService.java:39]: orElseThrow sends "Value not present" message. this is what i want to send actually. There is global exception handling mechanism so we do not need try catch block.

**Review by niloaydin — APPROVED** [review]

**Pull request merged**